### PR TITLE
docs:  glossary for important terms

### LINKED ISSUE #188: Glossary for important terms in `Safe-DS`
We should create a glossary of important terms that we use in the names of API elements and the documentation. On the one hand, this is a nice reference for users. On the other hand, it helps us ensure consistency in our design and documentation.

This can be akin to the [Google ML Glossary](https://developers.google.com/machine-learning/glossary).

Use [this](https://squidfunk.github.io/mkdocs-material/reference/tooltips/#adding-a-glossary) to implement the glossary.

### PR DESCRIPTION
**Pull request by patrikguempel**
Closes #188.

### Summary of Changes

Configured Snippets extension in mkdocs.yml
Created includes/glossary.md
Filled Glossary with various terms

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/6.22.2) status: ✅ [SUCCESS](https://github.com/Safe-DS/Stdlib/actions/runs/4897395746)

| Descriptor  |                                           Linter                                           |Files|Fixed|Errors|Elapsed time|
|-------------|--------------------------------------------------------------------------------------------|-----|-----|------|------------|
|✅ MARKDOWN  |[markdown-link-check](https://megalinter.io/6.22.2/descriptors/markdown_markdown_link_check)|    1|     |     0|0.41s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/6.22.2/descriptors/repository_git_diff)                    |yes  |     |no    |0.03s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/4897395746)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

**Comment by lars-reimann** [comment]
Looks great! Some other ideas:

* Metric
* Underfitting
* Overfitting
* Classification
* Regularization
* Linear Regression
* Feature
* Target
* Sample

File: mkdocs.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ nav:
     - Data Visualization: tutorials/data_visualization.ipynb
     - Machine Learning: tutorials/machine_learning.ipynb
   - API Reference: reference/
+  - Glossary: docs/glossary.md
```
Comment (lars-reimann) [inline mkdocs.yml:14]: That link's broken.

File: includes/glossary.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+*[Supervised Learning]:
+Supervised Learning is a subcategory of ML. This approach uses algorithms to learn given data.
+Those Algorithms might be able to find hidden meaning in data - without being told where to look.
+
+*[Tagged Table]:
+In addition to a regular table, a Tagged Table will mark one column as tagged, meaning that
+an applied algorithm will train to predict its entries.
+
+*[Regression]:
+Regression refers to the estimation of continuous dependent variables.
```
Comment (lars-reimann) [inline includes/glossary.md:30]: Would be nice to include the formula here (using LaTeX's block math mode)

File: includes/glossary.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+*[Precision]:
+The ability of a classification model to identify only the relevant data points.
```
Comment (lars-reimann) [inline includes/glossary.md:33]: Ditto

File: includes/glossary.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+*[Recall]:
+The ability of a classification model to identify all the relevant data points.
```
Comment (lars-reimann) [inline includes/glossary.md:36]: Ditto

File: includes/glossary.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+*[Accuracy]:
+The fraction of predictions a classification model has correctly identified.
```
Comment (lars-reimann) [inline includes/glossary.md:39]: Ditto

File: docs/glossary.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+## Feature
+Each feature represents a measurable piece of data that can be used for analysis.
+It is analogous to a column within a table.
+
+## Linear Regression
+Linear Regression is the supervised Machine Learning model in which the model finds the best fit linear line between the independent and dependent variable
+i.e. it finds the linear relationship between the dependent and independent variable.
+
+Implemented in Safe-DS as [LinearRegression][safeds.ml.classical.regression.LinearRegression].
```
Comment (lars-reimann) [inline docs/glossary.md:44]: Where applicable, please link to the documentation for the corresponding API element, as done here.

File: docs/glossary.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+## Confusion Matrix
+A confusion matrix is a table that is used to define the performance of a classification algorithm.
+It classifies the predictions to be either be true positive, true negative, false positive or false negative.
+
+## Decision Tree
+A Decision Tree represents the process of conditional evaluation in a tree diagram.
+
+## F1-Score
+The harmonic mean of [precision](#precision) and [recall](#recall). Formula:
```
Comment (lars-reimann) [inline docs/glossary.md:24]: If you use another term defined in the glossary, please link to its section, as done here.

**Review by lars-reimann — CHANGES_REQUESTED** [review]
Looks great and it's a very nice starting point for future additions. I'd like some more cross-references in the documentation (see comments), then this can be merged.

File: docs/glossary.md (hunk 7 above)
Comment (patrikguempel) [reply docs/glossary.md:27]: Done. Some references I put directly within the LaTeX code... please let me know if this works as intended.

Comment (lars-reimann) [reply docs/glossary.md:27]: You can check the generate page [here](https://safe-ds-stdlib--292.org.readthedocs.build/en/292/glossary/). As you can see there, placing references inside the LaTeX code does **not** work.

**Review by lars-reimann — APPROVED** [review]

**Pull request merged**